cmd/hostd/main.go

```diff
@@ -350,16 +357,36 @@ func main() {
 	}
 	defer rhp3WSListener.Close()
 
-	node, hostKey, err := newNode(walletKey, log)
+	node, hostKey, err := newNode(ctx, walletKey, log)
 	if err != nil {
 		log.Fatal("failed to create node", zap.Error(err))
 	}
 	defer node.Close()
 
+	opts := []api.ServerOption{
+		api.ServerWithAlerts(node.a),
+		api.ServerWithWebHooks(node.wh),
+		api.ServerWithSyncer(node.g),
+		api.ServerWithChainManager(node.cm),
+		api.ServerWithTransactionPool(node.tp),
+		api.ServerWithContractManager(node.contracts),
+		api.ServerWithAccountManager(node.accounts),
+		api.ServerWithVolumeManager(node.storage),
+		api.ServerWithRHPSessionReporter(node.sessions),
+		api.ServerWithMetricManager(node.metrics),
+		api.ServerWithSettings(node.settings),
+		api.ServerWithWallet(node.w),
+		api.ServerWithLogger(log.Named("api")),
+	}
+	if node.pinned != nil {
+		// pinner should be nil if explorer data is disabled
```

Comment:
I don't understand this comment? There doesn't seem to be any checks here that check explorer data being disabled yes or no? I think the `if` is not really necessary here, if `node.pinned` is `nil` we can safely set it on the `api` object.